transport/http/src/test/java/org/kaazing/gateway/transport/http/connector/specification/rfc7230/ConnectionManagementIT.java

```diff
@@ -0,0 +1,450 @@
+                        return null;
+                    }
+                });
+            }
+        });
+
+        connector.connect("http://localhost:8080/path", handler, new ConnectSessionInitializerGetConnection());
+        assertTrue(closed.await(2, SECONDS));
+
+        k3po.finish();
+    }
+
+    @Ignore("Doesn't Accept a second connection.")
+    @Test
+    @Specification({"client.must.not.reuse.tcp.connection.when.receives.connection.close/response"})
+    public void clientMustNotReuseTcpConnectionWhenReceivesConnectionClose() throws Exception {
+        final IoHandler handler = context.mock(IoHandler.class);
+        final CountDownLatch closed = new CountDownLatch(1);
+        context.checking(new Expectations() {
+            {
+                oneOf(handler).sessionCreated(with(any(IoSessionEx.class)));
+                oneOf(handler).sessionOpened(with(any(IoSessionEx.class)));
+                oneOf(handler).sessionClosed(with(any(IoSessionEx.class)));
+                will(new CustomAction("Latch countdown") {
+                    @Override
+                    public Object invoke(Invocation invocation) throws Throwable {
+                        closed.countDown();
+                        return null;
+                    }
+                });
+            }
+        });
+        connector.connect("http://localhost:8080/path1", handler, new ConnectSessionInitializerGetConnectionLength1());
+        closed.await(2, SECONDS);
+        final CountDownLatch closed2 = new CountDownLatch(1);
+        context.checking(new Expectations() {
+            {
+                oneOf(handler).sessionCreated(with(any(IoSessionEx.class)));
+                oneOf(handler).sessionOpened(with(any(IoSessionEx.class)));
+                oneOf(handler).sessionClosed(with(any(IoSessionEx.class)));
+                will(new CustomAction("Latch countdown") {
+                    @Override
+                    public Object invoke(Invocation invocation) throws Throwable {
+                        closed2.countDown();
+                        return null;
+                    }
+                });
+            }
+        });
+        connector.connect("http://localhost:8080/path2", handler, new ConnectSessionInitializerGetConnectionLength2());
+        closed2.await(2, SECONDS);
+
+        k3po.finish();
+    }
+
+    @Test
+    @Specification({"server.getting.upgrade.request.must.respond.with.upgrade.header/response"})
+    public void serverGettingUpgradeRequestMustRespondWithUpgradeHeader() throws Exception {
+        final IoHandler handler = context.mock(IoHandler.class);
+        final CountDownLatch closed = new CountDownLatch(1);
+
+        context.checking(new Expectations() {
+            {
+                oneOf(handler).sessionCreated(with(any(IoSessionEx.class)));
+                oneOf(handler).sessionOpened(with(any(IoSessionEx.class)));
+                oneOf(handler).sessionClosed(with(any(IoSessionEx.class)));
```

Comment:
I'm thinking maybe we can check if a message is received, and that the message is what we are expecting from the origin server (for example if we get upgrade header).